src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -738,52 +738,61 @@ public static TimeSpan GetTotalPauseDuration()
             return new TimeSpan(_GetTotalPauseDuration());
         }
 
-        internal delegate void EnumerateConfigurationValuesCallback(int type, long intValue, bool boolValue);
+        [ThreadStatic]
+        private static Dictionary<string, object>? _result;
 
-        /// <summary>
-        /// Returns a dictionary keyed on GC Configuration Names with values as the configuration values.
-        /// </summary>
-        /// <remarks>
-        /// Corresponding issue: https://github.com/dotnet/runtime/issues/67471
-        /// </remarks>
-        public static unsafe IReadOnlyDictionary<string, object> GetConfigurationVariables()
+        [UnmanagedCallersOnly]
+        private static unsafe void Callback(void* name, GCConfigurationType type, long data)
         {
-            Dictionary<string, object> result = new Dictionary<string, object>();
+            if (name == null || _result == null)
+            {
+                return;
+            }
 
-            string? name  = null;
-            string? value = null;
+            string? nameAsString = Marshal.PtrToStringUTF8((IntPtr)name);
```

Comment:
If we have a non-null input, what is the case where `Marshal.PtrToStringUTF8()` will return null?